roles/confluent.common/tasks/remove_packages.yml

```diff
@@ -26,11 +26,27 @@
     msg: "{{ confluent_packages_removed }}"
   when: confluent_packages_removed|length > 0
 
+### Kafka Connect Replicator Binary does not include a service definition by default, thus we need to validate if one exists in order to handle service lifecycle correctly.
+- name: Get Service Facts
+  ansible.builtin.service_facts:
+
+- name: Check "{{ service_name }}" exists in SystemD
+  set_fact:
+    confluent_services_actual: >
+      {{ ansible_facts.services | dict2items | map(attribute='key') | select('match', '{{ service_name }}') | list }}
+
+- name: Debug Confluent Services
+  debug:
+    msg: "{{ confluent_services_actual }}"
+  when: confluent_services_actual|length > 0
+
 - name: Stop Service before Removing Confluent Packages
   systemd:
     name: "{{ service_name }}"
     state: stopped
-  when: confluent_packages_removed|length > 0
+  when:
+    - confluent_services_actual|length > 0
+    - confluent_packages_removed|length > 0
```

Comment:
Hey great approach here confirming the systemd service exists using the `service_facts` module- before stopping the service. I'm wondering 2 things

1) Do we need the `Get Service Facts` task in line 30 if the same task gets run earlier here: https://github.com/JumaX/cp-ansible/blob/package_removal_update/kafka_connect_replicator.yml#L11

Running it twice might be the safe play if we are assuming users use our roles without our playbooks... so keeping it is probably fine. lmk your thoughts.

2) I've debugged the `ansible_facts.services` variable, and it appends `.service` to the end of the service name like so:
```
ok: [kafka-connect-replicator1] => {
    "ansible_facts.services": {
        "autovt@.service": {
            "name": "autovt@.service",
            "source": "systemd",
            "state": "unknown",
            "status": "disabled"
        },
        "confluent-control-center.service": {
            "name": "confluent-control-center.service",
            "source": "systemd",
            "state": "inactive",
            "status": "disabled"
        },
```
I'm wondering if you can simplify the logic to 
```suggestion
  when:
      - ansible_facts.services[service_name + '.service'] is defined
      - confluent_packages_removed|length > 0
```

Reply:
Good points.

1. I don't think we need the second check.  I did not see the first thus why I included it.  We don't currently support running as individual tasks outside the playbook so I think it makes sense to remove.

2. This is far more elegant and I have updated the code accordingly.  Good catch!